includes/discovery/vrf.inc.php

```diff
@@ -140,45 +140,90 @@
                 }
             }//end if
         }//end foreach
+    } else if ($device['os_group'] == 'nokia') {
+        unset($vrf_count);
 
-        unset(
-            $descr_table,
-            $port_table
-        );
-        echo "\n";
-
-        $sql = "SELECT * FROM ports WHERE device_id = '".$device['device_id']."'";
-        foreach (dbFetchRows($sql) as $row) {
-            $if     = $row['port_id'];
-            $vrf_id = $row['ifVrf'];
-            if ($row['ifVrf']) {
-                if (!$valid_vrf_if[$vrf_id][$if]) {
-                    echo '-';
-                    dbUpdate(array('ifVrf' => 'NULL'), 'ports', 'port_id=?', array($if));
-                } else {
-                    echo '.';
-                }
+        $vrtr = snmpwalk_cache_oid($device, 'vRtrConfTable', array(), 'TIMETRA-VRTR-MIB', null);
+        $port_table = snmpwalk_cache_twopart_oid($device, 'vRtrIfName', array(), 'TIMETRA-VRTR-MIB', null);
```

Comment:
Scrutinizer: "The call to ``snmpwalk_cache_twopart_oid()`` has too many arguments starting with ``null``."